### ISSUE TITLE
VirtualEnv vs. Conda

### ISSUE THREAD

**Issue by gblinick** [Satalia/production-data-science#23]
Hi @FilippoBovo ,

I finished your tutorial. Thanks for writing it. It was well written and informative.

I am wondering though, why you chose to create a virtual environment based on virtualenv instead of conda. In my own limited experience doing Data Science, I have always used Anaconda and didn't come across using virtualenvs until this tutorial. I also found [this article ](https://medium.com/@__pamaron__/understanding-and-use-python-virtualenvs-from-data-scientist-perspective-bfed61faeb3f) comparing the two where the author recommends using Anaconda. 

Is the reason because virtualenvs are far more common in the general software engineering world and you wanted to use this more general practice?

Thanks!

**Comment by FilippoBovo** [comment]
Hi @gblinick,

Thank you. I am happy that you found the tutorial informative. I hope you will find it useful, as well.

Regarding the choice of virtualenv, it comes down to my experience and the experience of some colleagues. In the past, I switched from pip+virtualenv to conda twice, but got back to pip+virtualenv, eventually.

On one hand, the package manager experience was better in conda. On the other hand, when I needed to install a package that was not in the conda repository, I had to use pip, and most of the time the package installed with pip would not work well due to the lack of compatibility between conda and pip dependencies. This is the reason why I reverted back to pip+virtualenv.

Note that I tried conda one and a half years ago, and things may have changed now.
